.github/workflows/publish_snapshot.yml

```diff
@@ -0,0 +1,33 @@
+name: Publish a snapshot release
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  publish_snapshot:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
+
+      - uses: actions/setup-java@d854b6da19cdadd9a010605529e522c2393ebd38
+        with:
+          distribution: 'zulu'
+          java-version: 17
+
+      - uses: gradle/wrapper-validation-action@e6e38bacfdf1a337459f332974bb2327a31aaf4b
+
+      - name: Setup Gradle
+        uses: gradle/gradle-build-action@356abb47e7664b5505e25d7997a5a522a17c62d9
+        with:
+          gradle-version: wrapper
+
+      - name: Publish the artifacts
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          ORG_GRADLE_PROJECT_mavenCentralUsername: ${{ secrets.SONATYPE_NEXUS_USERNAME }}
+          ORG_GRADLE_PROJECT_mavenCentralPassword: ${{ secrets.SONATYPE_NEXUS_PASSWORD }}
+          ORG_GRADLE_PROJECT_signingInMemoryKey: ${{ secrets.ARTIFACT_SIGNING_PRIVATE_KEY }}
```

Comment:
thanks a lot for these! My gradle properties also specifies signing keyId and signing password. I'm assuming I'll need those too if you can add them in quickly. I'll add those secrets to the repo before I merge this so we can give it a shot.

Appreciate it!

Reply:
With `signingInMemoryKey` you don't need the key id anymore (since the actual signing key itself is used). I will add a property for the password. 

I will also add a property that automatically closes and releases the sonatype nexus repo so that it's a completely unattended process (other than merging `master` into `release`)

Reply:
Updated with the above changes

Reply:
does snapshot need the password for the key too?

Reply:
No, snapshots don't need to be signed.

Reply:
Oh, maybe they do? I'll double check.

Reply:
yea, they don't. I haven't been signing them before but I saw you included the key here too so I wanted to check if I'll need the password then as well

Reply:
Whoops, I confused myself. I'll fix this later today.

Reply:
OK should be ready to go now